cli/src/klio_cli/commands/job/gke.py

```diff
@@ -52,110 +54,114 @@ class GKECommandMixin(object):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._deployment_config = None
-        self._kubernetes_client = None
+        self._dynamic_client = None
         self._kubernetes_active_context = None
 
     @property
-    def kubernetes_client(self):
-        if not self._kubernetes_client:
+    def kubernetes_active_context(self):
+        if not self._kubernetes_active_context:
             # TODO: This grabs configs from '~/.kube/config'. @shireenk
             #  We should add a check that this file exists
             # If it does not exist then we should create configurations.
             # See link:
             # https://github.com/kubernetes-client/python-base/blob/master/config/kube_config.py#L825
             k8s_config.load_kube_config()
-            self._kubernetes_client = k8s_client.AppsV1Api()
-
-        return self._kubernetes_client
-
-    @property
-    def kubernetes_active_context(self):
-        if not self._kubernetes_active_context:
             _, active_context = k8s_config.list_kube_config_contexts()
             self._kubernetes_active_context = active_context
 
         return self._kubernetes_active_context
 
     @property
-    def deployment_config(self):
-        if not self._deployment_config:
-            path_to_deployment_config = os.path.join(
-                self.job_dir, "kubernetes", "deployment.yaml"
-            )
-            with open(path_to_deployment_config) as f:
-                self._deployment_config = yaml.safe_load(f)
-        return self._deployment_config
+    def dynamic_client(self):
+        if not self._dynamic_client:
+            k8s_config.load_kube_config()
+            with k8s_client.ApiClient() as api_client:
+                dyn_client = DynamicClient(api_client)
+                self._dynamic_client = dyn_client
+        return self._dynamic_client
 
-    def _deployment_exists(self):
-        """Check to see if a deployment already exists
+    def get_deployment_config(self, config_dir="kubernetes"):
+        """Searches kubernetes directory for a Deployment
```

Comment:
I wonder if (maybe in a subsequent PR) we should do something more comprehensive if there are actually multiple deployment files.  

I guess, does it make sense/is it not uncommon to have multiple such files?  If so, maybe we should require a specifier or be more explicit about which one is used by default.  If not, maybe we should error if multiple are detected.  Also maybe this is overkill so I'm also fine with not doing anything if others agree.

Reply:
Yeah, for this PR i assumed that we deploy and delete the other resources like hpa and vpa but we interact with Deployments much more. which is why i made that `get_deployment_config` but if we find ourselves interacting with the other resources we can do something like `get_resource_config(resource_type="Deployment")`.
But leaning towards crossing that bridge when we see how we/users will interact with the other resources